plugin.audio.soundcloud/resources/plugin.py

```diff
@@ -0,0 +1,146 @@
+from future import standard_library
+standard_library.install_aliases()  # noqa: E402
+
+from resources.lib.soundcloud.api_v2 import ApiV2
+from resources.lib.kodi.cache import Cache
+from resources.lib.kodi.items import Items
+from resources.lib.kodi.search_history import SearchHistory
+from resources.lib.kodi.settings import Settings
+from resources.lib.kodi.vfs import VFS
+from resources.routes import *
+import urllib.parse
+import sys
+import xbmc
+import xbmcaddon
+import xbmcgui
+import xbmcplugin
+
+addon = xbmcaddon.Addon()
+addon_id = addon.getAddonInfo("id")
+addon_base = "plugin://" + addon_id
+addon_profile_path = xbmc.translatePath(addon.getAddonInfo('profile')).decode('utf-8')
```

Comment:
`.decode('utf-8')` is going to break in Python 3. I'd recommend to use kodi-six: https://github.com/romanvm/kodi.six

Reply:
Good to know! Did I miss this in the docs?
I got this from the wiki, which should probably be updated: https://kodi.wiki/view/Add-on_rules

Reply:
This page should also contain some infos on Python 3 compatibility: https://kodi.wiki/view/Add-on_unicode_paths